File: src/main/java/com/foodrescue/service/RestaurantService.java
package com.foodrescue.service;

import java.util.List;

import com.foodrescue.dto.RestaurantDTO;

public interface RestaurantService {

    /**
     * Register a new restaurant
     *
     * @param restaurantDTO
     * @return RestaurantDTO
     */
    RestaurantDTO registerRestaurant(RestaurantDTO restaurantDTO);

    /**
     * Get all restaurants
     *
     * @return List<RestaurantDTO>
     */
    List<RestaurantDTO> getAllRestaurants();

    /**
     * Get restaurant by ID
     *
     * @param restaurantId
     * @return RestaurantDTO
     */
    RestaurantDTO getRestaurantById(Long restaurantId);

    /**
     * Update restaurant details
     *
     * @param restaurantId
     * @param restaurantDTO
     * @return RestaurantDTO
     */
    RestaurantDTO updateRestaurant(Long restaurantId,
                                   RestaurantDTO restaurantDTO);

    /**
     * Delete restaurant
     *
     * @param restaurantId
     */
    void deleteRestaurant(Long restaurantId);

    /**
     * Get restaurant by email
     *
     * @param email
     * @return RestaurantDTO
     */
    boolean existsByPhone(String phone);

    /**
     * Check whether license already exists
     *
     * @param licenseNumber
     * @return boolean
     */
    boolean existsByLicenseNumber(String licenseNumber);

}

File: src/services/restaurantService.js
import api from './api';

export const restaurantService = {
  registerRestaurant: async (restaurantDTO) => {
    const response = await api.post('/api/restaurants/register', restaurantDTO);
    return response.data;
  },

  getAllRestaurants: async () => {
    const response = await api.get('/api/restaurants');
    return response.data;
  },

  getRestaurantById: async (id) => {
    const response = await api.get(`/api/restaurants/${id}`);
    return response.data;
  },

  updateRestaurant: async (id, restaurantDTO) => {
    const response = await api.put(`/api/restaurants/${id}`, restaurantDTO);
    return response.data;
  },

  deleteRestaurant: async (id) => {
    const response = await api.delete(`/api/restaurants/${id}`);
    return response.data;
  },
};

export default restaurantService;
